cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1351,14 +1355,26 @@ func (p *BicepProvider) createOutputParameters(
 	return outputParams
 }
 
+type bicepModule struct {
+	fileType bicepModuleType
+	filePath string
+}
+
 // loadParameters reads the parameters file template for environment/module specified by Options,
 // doing environment and command substitutions, and returns the values.
-func (p *BicepProvider) loadParameters(ctx context.Context) (map[string]azure.ArmParameterValue, error) {
-	parametersTemplateFilePath := p.parametersTemplateFilePath()
-	log.Printf("Reading parameters template file from: %s", parametersTemplateFilePath)
-	parametersBytes, err := os.ReadFile(parametersTemplateFilePath)
+func (p *BicepProvider) loadParameters(
```

Comment:
My preference would be that this function continue to just read the `.parameters.json` function, with the expectation that it not be called in the `.bicepparam` case vs having the check in this function.  That would allow the signature to go back to as it was.

You already ensure you don't call this function in the `.bicepparam` case in the one caller (`plan`).